cdci_data_analysis/plugins/dummy_plugin/empty_instrument_with_posix_path.py

```diff
@@ -0,0 +1,144 @@
+"""
+Overview
+--------
+   
+general info about this module
+
+
+Classes and Inheritance Structure
+----------------------------------------------
+.. inheritance-diagram:: 
+
+Summary
+---------
+.. autosummary::
+   list of the module you want
+    
+Module API
+----------
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__author__ = "Andrea Tramacere"
+
+# Standard library
+# eg copy
+# absolute import rg:from copy import deepcopy
+
+# Dependencies
+# eg numpy 
+# absolute import eg: import numpy as np
+
+# Project
+# relative import eg: from .mod import f
+
+
+from cdci_data_analysis.analysis.instrument import Instrument
+from cdci_data_analysis.analysis.queries import SourceQuery, InstrumentQuery, Float
+
+from .data_server_dispatcher import (EmptyProductQuery, 
+                                     DataServerNumericQuery, 
+                                     FailingProductQuery, 
+                                     DataServerParametricQuery, 
+                                     EchoProductQuery,
+                                     DefaultEchoProductQuery,
+                                     FileParameterQuery)
+
+# duplicated with jemx, but this staticmethod makes it complex.
+# this all should be done commonly, for all parameters - limits are common thing
+from ...analysis.exceptions import RequestNotUnderstood
+from ...analysis.parameters import (SpectralBoundary, 
+                                    Angle, 
+                                    Energy, 
+                                    Integer, 
+                                    Float, 
+                                    String,
+                                    StructuredParameter,
+                                    POSIXPath)
+
+
+
+class BoundaryFloat(Float):
+    def check_value(self):
+        super().check_value()
+        if self.value > 800:
+            raise RequestNotUnderstood('p value is restricted to 800 W')
+
+
+def my_instr_factory():
+    src_query = SourceQuery('src_query')
+
+    # empty query
+    instr_query = InstrumentQuery(name='empty_instrument_query',
+                                  input_prod_list_name='p_list',
+                                  catalog=None,
+                                  catalog_name='user_catalog')
+
+    empty_query = EmptyProductQuery('empty_with_posix_path_parameters_dummy_query',)
+    failing_query = FailingProductQuery('failing_parameters_dummy_query',)
+    # let's build a simple parameter to its list
+    p = BoundaryFloat(value=10., name='p', units='W',)
+    numerical_query = DataServerNumericQuery('numerical_parameters_dummy_query',
+                                             parameters_list=[p])
+
+    f = POSIXPath(value=None, name='dummy_POSIX_file', is_optional=True)
```

Comment:
Do we really need all other parameters/queries in this dummy instrument?

Reply:
not really, I just copied from another instrument